src/stats.jl

```diff
@@ -171,6 +171,15 @@ function describe(
     return dfs
 end
 
+function Base.show(io::IO, mime::MIME"text/plain", cs::Vector{ChainDataFrame})
```

Comment:
This only captures the abstract `Vector{ChainDataFrame}` but not any concretely typed `Vector{<:ChainDataFrame}`. In general, I am a bit worried about changing the display of vectors of `ChainDataFrame`s - it seems wrong to completely opt out of the default display mechanism of vectors in Julia (I also wonder if it causes any problems) just to change the way in which `describe(chain)` is displayed. Maybe rather `describe` should not return a vector of `ChainDataFrame`.

Reply:
Yeah, I'm down with this. My understanding of your suggestion is that `describe` should be a pure IO function -- we should make `display(io::IO, chn::Chain)` and the output is all the stuff inside the `Base.show` definition above.

Reply:
We should not define `display` (it calls `show(io, MIME("text/plain"), x)` which should be implemented), but we should just implement `DataAPI.describe(io, chain)` if we want to display the summary statistics in a nice way: https://juliastats.org/StatsBase.jl/latest/scalarstats/#Summary-Statistics-1

Reply:
Thank you so much for your help! So, to double check that I'm understanding these suggestions. Instead of returning a `Vector{ChainDataFrame}`,  `describe` should be an implementation of `StatsBase.describe` and return something like this?

```julia
chn = Chains(rand(100, 2, 2), [:a, :b])
chn_arr = Array(chn)
sections = chn.name_map[:parameters]
for i in 1:length(sections)
    println("Parameter $(sections[i])")
    describe(chn_arr[:,i])
    println()
end
```
![StatsBase _describe](https://user-images.githubusercontent.com/31251575/125004097-600c6d80-e01e-11eb-8af2-4475c52aad98.png)

Reply:
I think it's fine if `summarystats` returns different summary statistics (so this method does not have to be changed) but we should make sure that `describe` just prints these summary statistics in a pretty way to be consistent with how `describe` and `summarystats` are defined in StatsBase. I.e., in particular `describe` should not return anything but only print to IO and it should not print the quantiles if they are not part of `summarystats` (here it might actually be better to include them in `summarystats` as well and return two dataframes, possibly as a named tuple).

Reply:
To clarify maybe a little bit, there's no need to change to this code you've posted:

```julia
chn = Chains(rand(100, 2, 2), [:a, :b])
chn_arr = Array(chn)
sections = chn.name_map[:parameters]
for i in 1:length(sections)
    println("Parameter $(sections[i])")
    describe(chn_arr[:,i]) # <- This is not what we want, we want to print the results of `describe(chain)` here instead
    println()
end
```

Basically we want to change the stuff that is _currently_ in `show` to `describe`, so that `describe` becomes a pure IO function and not a weird `Vector{<:ChainDataFrame}` thing that we have now.

Maybe one way is to do something like

```julia
function DataAPI.describe(io::IO, chains::Chains)
    print(io, "Chains ", chains, ":\n\n", header(chains))

    summstats = summarystats(chains)
    qs = quantiles(chains)

    println(io)
    show(io, summstats)

    println(io)
    show(io, qs)
end
```

which won't actually return anything, it just prints stuff out to the screen. There's probably a way more sane way to do this, but it's a rough sketch to get you started.